inntekt/src/main/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/inntekt/MapInntekt.kt

```diff
@@ -3,14 +3,23 @@ package no.nav.helsearbeidsgiver.inntektsmelding.inntekt
 import no.nav.helsearbeidsgiver.felles.Inntekt
 import no.nav.helsearbeidsgiver.felles.MottattHistoriskInntekt
 import no.nav.helsearbeidsgiver.inntekt.InntektskomponentResponse
+import java.math.BigDecimal
 
-fun mapInntekt(response: InntektskomponentResponse): Inntekt {
+fun mapInntekt(response: InntektskomponentResponse, orgnr: String): Inntekt {
     val list = mutableListOf<MottattHistoriskInntekt>()
+    var sum = BigDecimal(0.0)
+    if (orgnr.isNullOrEmpty()) {
+        return Inntekt(sum.toDouble(), list)
+    }
     response.arbeidsInntektMaaned?.forEach { maaned ->
-        maaned.arbeidsInntektInformasjon?.inntektListe?.map { inntekt ->
-            list.add(MottattHistoriskInntekt(maaned.aarMaaned, inntekt.beloep ?: 0.0))
+        maaned.arbeidsInntektInformasjon?.inntektListe?.filter {
+            it.virksomhet?.identifikator.equals(orgnr)
+        }?.toList()?.map { inntekt ->
+            val belop = inntekt.beloep ?: 0.0
+            list.add(MottattHistoriskInntekt(maaned.aarMaaned, belop))
+            sum = sum.add(BigDecimal.valueOf(belop))
         }
     }
```

Comment:
Her ville jeg delt koden over flere linjer. Og blitt kvitt mutable (list og sum), for øvrig.